src/routes/room.js

```diff
@@ -1,17 +1,36 @@
 const router = require('express').Router();
 const StudyRoom = require('../models/studyRoom.model');
 const Student = require('../models/student.model');
-const CourseNotes = require('../models/courseNotes.model');
-const { createStudyRoom, editStudyRoom } = require("../factory/roomFactory");
-const createCourseNotes = require("../factory/courseNotesFactory");
+const { createStudyRoom } = require("../factory/roomFactory");
+const CourseNotesRepository = require("../repository/courseNotesRepository")
+const StudyRoomRepository = require('../repository/studyRoomRepository');
+const StudentRepository = require('../repository/studentRepository');
+const studyRoomRepository = require('../repository/studyRoomRepository');
 
 
 router.route('/').put((req, res) => {
-  StudyRoom.findOne({ studyRoomID: req.body.studyRoomID.toString() })
-    .then(() => {
-      const room = editStudyRoom(req.body)
-      room.save()
-        .then(() => res.json(`Room ${room.studyRoomID} updated`))
+  StudyRoomRepository.findOne(req.body.studyRoomID)
+    .then((room) => {
+      if (data.owner) {
+        room.owner = data.owner;
+      }
+      if (data.color) {
+        room.color = data.color;
+      }
+      if (data.description) {
+        room.description = data.description;
+      }
+      if (data.title) {
+        room.title = data.title;
+      }
+      if (data.avatarText) {
+        room.avatar = data.avatarText;
+      }
+      if (data.participants) {
+        room.participants = data.participants;
+      }
+      StudyRoomRepository.updateOne(room)
```

Comment:
Method `updateOne(...)` does not exist in `StudyRoomRepository`. For updating a room, the method is called 'save(...)`. Either change this line to `StudyRoomRepository.save(...)` or rename the method in `StudyRoomRepository` to `updateOne(...)`.